www/docs/getting-started/client.md

```diff
@@ -328,6 +328,24 @@ export default function App ({ Component, pageProps }) {
 
 If you pass the `session` page prop to the `<Provider>` – as in the example above – you can avoid checking the session twice on pages that support both server and client side rendering.
 
+This only works on pages where you provide the correct `pageProps`, however. This is normally done in `getInitialProps` or `getServerSideProps` like so:
+
+```js title="pages/index.js"
+import { getSession } from "next-auth/client"
+
+...
+
+export async function getServerSideProps(ctx) {
+  return {
+    props: {
+      session: await getSession(ctx)
+    }
+  }
+}
+```
+
+If every one of your pages needs to be protected, you can do this in `_app`, otherwise you can do it on a page-by-page basis.
```

Comment:
Don't know if it's a true alternative, but the method described here https://github.com/nextauthjs/next-auth/issues/1210#issuecomment-782630909 makes it possible to do per page auth much easier without gSSP, and you can even show a fallback ui or extend it with roles or anything really.

gSSP in _app always felt very harsh to me, but I can see it's easier. What do you think?

Reply:
So the method you've linked looks really easy to implement with some nice conveniences tacked on, but I have to admit im not familiar enough with our internal session handling currently to fully appreciate the difference haha

I can definitely mention it in the docs though :+1:

Reply:
So it boils down to that if you use gSSP on the _app component, it means EVERY page transition will ALWAYS be blocking (meaning SSR), versus my suggestion, first you show a skeleton while loading/checking auth, and every page transition will be client-side afterwards.